hyperspy/io_plugins/attolight.py

```diff
@@ -0,0 +1,351 @@
+
+    # Estimate start and end wavelengths
+    spectra_offset_array = [central_wavelength - 273, central_wavelength + 273]
+
+    # Apply calibration
+    size = data.shape[0]
+    name = 'Wavelength'
+    scale = (spectra_offset_array[1] - spectra_offset_array[0]) \
+            / size
+    offset = spectra_offset_array[0]
+    units = '$nm$'
+
+    axis_dict = {'name': name,
+                 'units': units,
+                 'navigate': False,
+                 'size': size,
+                 'scale': scale,
+                 'offset': offset}
+
+    return axis_dict
+
+
+def _create_navigation_axis(data, metadata):
+    # Edit the navigation axes
+
+    # Get relevant parameters from metadata and acquisition_systems
+    # parameters
+    acquisition_system = metadata.Acquisition_instrument.acquisition_system
+    cal_factor_x_axis \
+        = attolight_systems[acquisition_system]['cal_factor_x_axis']
+    FOV = metadata.Acquisition_instrument.SEM.FOV
+    nx = metadata.Acquisition_instrument.SEM.resolution_x
+
+    # Get the calibrated scanning axis scale from the acquisition_systems
+    # dictionary
+    calax = cal_factor_x_axis / (FOV * nx)
+
+    size = data.shape[1]
+    name = ''
+    scale = calax * 1000
+    # changes micrometer to nm, value for the size of 1 pixel
+    units = '$nm$'
+
+    axis_dict = {'units': units,
+                 'navigate': True,
+                 'size': size,
+                 'scale': scale, }
+
+    return axis_dict
+
+
+def _save_background_metadata(metadata, hypcard_folder, background_file_name='Background*.txt'):
+    """
+    Based on the Attolight background savefunction.
+    If background is found in the folder, it saves background as in the metadata.
+    """
+    # Get the absolute path
+    path = os.path.join(hypcard_folder, background_file_name)
+
+    # Try to load the file, if it exists.
+    try:
+        # Find the exact filename, using the * wildcard
+        path = glob.glob(path)[0]
+        # Load the file as a numpy array
+        bkg = np.loadtxt(path)
+        # The bkg file contains [wavelength, background]
+        metadata.set_item("Signal.background", bkg)
+        return
+    except:
+        return
+
+
+def file_reader(filename, attolight_acquisition_system='cambridge_uk_attolight',
+                *args, **kwds):
+    """Loads data into CLSEMSpectrum lumispy object.
+    Reads the HYPCard.bin file, containing the hyperspectral CL data.
+    Reads the metadata files contained in the same folder as HYPCard.bin.
+        - MicroscopeStatus.txt to load all the metadata values
+        - Background file contained in the same folder
+        - SEM image (taken simultaneously) with the hyperspectral map.
+
+    Parameters
+    ----------
+    filename : str, None
+        The HYPCard.bin filepath for the file to be loaded, created by
+        the AttoLight software.
+        If None, a pop-up window will be loaded.
```

Comment:
Remove this line, since this is not correct - this is not what this function do.